### PR TITLE
Make spektrum_rc ready to be used in other boards

### PR DESCRIPTION
**Pull request by zehortigoza**

### CODE REVIEW

**Comment by zehortigoza** [comment]
@lucasdemarchi

File: src/drivers/boards/eagle/board_config.h [hunk 1]
```diff
@@ -42,3 +42,6 @@
 #define BOARD_NAME "EAGLE"
 #define BOARD_BATTERY1_V_DIV   (10.177939394f)
 #define BOARD_BATTERY1_A_PER_V (15.391030303f)
+
+// Snapdraogon: use J15 (next to USB)
```
Comment (lucasdemarchi) [inline src/drivers/boards/eagle/board_config.h]: while at it, fix the typo?

File: src/drivers/spektrum_rc/spektrum_rc.cpp [hunk 2]
```diff
@@ -89,17 +90,35 @@ void task_main(int argc, char *argv[])
...
+	fds[0].fd = uart_fd;
+	fds[0].events = POLLIN;
+#endif
+
 	// Main loop
 	while (!_task_should_exit) {
 
-		int newbytes = ::read(uart_fd, &rx_buf[0], sizeof(rx_buf));
+#if !defined(__PX4_QURT)
+		int ret = poll(fds, sizeof(fds) / sizeof(fds[0]), -1);
```
Comment (lucasdemarchi) [inline src/drivers/spektrum_rc/spektrum_rc.cpp]: you are using only 1 fd... I supposed you are using poll() + usleep() here to simulate vmin/vtim that nuttx doesn't have. Couldn't you do the same by just making the fd a blocking descriptor? Here you are making a blocking poll() anyway.

**Comment by lucasdemarchi** [comment]
Other than the comments above, LGTM... however I'm not convinced about using it rather than the better tested and working implementation using the fpu driver. I'd nice to have this cleaner implementation, but it has some cons:

1. It makes it more difficult to have other serial protocols - we would need to implement a similar mechanism of what's currently on fmu.cpp
1. The implementation is used by only one board and thus doesn't share the test coverage that the fpu implementation has

File: src/drivers/spektrum_rc/spektrum_rc.cpp (hunk 2 above)
Comment (zehortigoza) [inline src/drivers/spektrum_rc/spektrum_rc.cpp]: No the `usleep(10000);` is for qurt boards that do not have poll.

**Comment by zehortigoza** [comment]
> Other than the comments above, LGTM... however I'm not convinced about using it rather than the better tested and working implementation using the fpu driver. I'd nice to have this cleaner implementation, but it has some cons:
> 1. It makes it more difficult to have other serial protocols - we would need to implement a similar mechanism of what's currently on fmu.cpp

When we need another protocols we can always go back to fmu or have a new driver but right now Aero RTF only needs DSM so it is saving resources that would be spent when adding run-time support to another protocols.

> 2. The implementation is used by only one board and thus doesn't share the test coverage that the fpu implementation has

The important part the `dsm_parse()` is shared between fmu and spektrum_rc.

File: src/drivers/spektrum_rc/spektrum_rc.cpp (hunk 2 above)
Comment (lucasdemarchi) [inline src/drivers/spektrum_rc/spektrum_rc.cpp]: and you wouldn't need poll() if you consider the file descriptor as blocking... so you wouldn't need the different implementations

**Comment by lucasdemarchi** [comment]
> The important part the dsm_parse() is shared between fmu and spektrum_rc.
> Add your review

yep, but my comment remains.... I don't see a very compelling reason to do it, particularly when a new release is around the corner.

**Comment by zehortigoza** [comment]
Ok I will remove the last commit.

**Comment by lucasdemarchi** [comment]
> Ok I will remove the last commit.

the comment for the unneeded difference with qurt remains: for the current usage, with a single fd, you don't need poll.

**Comment by zehortigoza** [comment]
> the comment for the unneeded difference with qurt remains: for the current usage, with a single fd, you don't need poll.

I think is better poll than awake process to do nothing. Also there is DSM receivers that send data at each 11ms and others at each 22ms this would be beneficial when using the one that updates at each 22ms.

**Pull request closed without merging**